### DISCUSSION TITLE
Warn upon upgrading to development version how not to lose years of blocking

### DISCUSSION THREAD

**Discussion by jidanni** [uBlockOrigin/uBlock-issues#1835] [General]
User updates from
https://chrome.google.com/webstore/detail/ublock-origin/cjpalhdlnbpafiamejdnhcphjbkeiagm to
https://chrome.google.com/webstore/detail/ublock-origin-development/cgbcahbpdhpcegmbfconppldiemgcoii

User loses all his years of site blocking settings!

Therefore please on
https://chrome.google.com/webstore/detail/ublock-origin-development/cgbcahbpdhpcegmbfconppldiemgcoii
add a warning to do https://www.ghacks.net/2018/02/16/ublock-origin-development-add-on-for-firefox/ steps
> Select the "back up to file"option under Settings, and save the text document to the local system.

etc. before it is too late. Thanks.
(Apparently "Warn upon upgrading to development version" would not be that easy, so warnings need to be in the instructions.)

**Comment by ghost** [comment]
>User loses all his years of site blocking settings!

Why? Both version will run simultaneously. Disable the other one. If you want the settings - copy them or backup-restore.

**Reply by jidanni** [reply]
Yes, but many users will just
- Remove the stable version of uBlock Origin
- Install the development version of uBlock Origin instead

And then it is too late!

![20211123T075136](https://user-images.githubusercontent.com/1230959/142951978-4c4a1827-679d-4265-900d-1b2724563508.jpg)

I.e., you are counting on Google always showing step 3.
Therefore you should provide instructions of your own, there on the Chrome extension store page.

**Reply by u-RraaLL** [reply]
These are steps for Firefox, not Chrome. And they are inaccurate.

No need to delete the stable version. It'll update to dev build. On browser restart or on manually clicking the "update" button in about:addons (after trying to install the dev build).

**Reply by jidanni** [reply]
Well whatever the steps may be, please publish them on
https://chrome.google.com/webstore/detail/ublock-origin-development/cgbcahbpdhpcegmbfconppldiemgcoii
Thanks.

**Comment by uBlock-user** [comment]
> User loses all his years of site blocking settings!

Been a uBO user since 2016, never lost any settings upon upgrading to dev build, so where are you getting this mis-information ?


> Remove the stable version of uBlock Origin

Where is this written ?

**Reply by jidanni** [reply]
So we just install the dev version,
and remove the non-dev version.

Or

Remove the non-dev version,
and install the dev-version,

and our settings will be grandfathered?
Not in the default case!

So I'm saying something needs to be mentioned somewhere.

**Reply by uBlock-user** [reply]
You install the dev build and Firefox will overwrite that over stable build(hence stable build getting removed), but on Chrome you will have both extensions installed.

**Reply by jidanni** [reply]
I can guarantee that 99% of users will never expect such different behaviors on different systems.
So Chrome users will end up following Firefox users' blog posts.
And Firefox users' will end up following Chrome users' blog posts.
And some users will lose their settings.
And that's the way it will be, unless there is some official documentation provided.

**Reply by jidanni** [reply]
Well even a single user's loss of his settings is too much of a price to pay.
I would submit a pull request, but er, I haven't even enabled Advanced User Mode.
